schema/any.go

```diff
@@ -67,12 +67,12 @@ func (a *AnySchema) ValidateCompatibility(typeOrData any) error {
 	case reflect.Map:
 		return nil
 	default:
-		// Check if it's an actual 'any' schema.
-		schemaType, ok := typeOrData.(*AnySchema)
-		if !ok {
+		switch typeOrData.(type) {
+		case *AnySchema, *OneOfSchema[int64], *OneOfSchema[string]:
+		default:
 			// It's not an any schema, so error
 			return &ConstraintError{
-				Message: fmt.Sprintf("unsupported schema type for 'any' type: %T", schemaType),
+				Message: fmt.Sprintf("unsupported schema type for 'any' type: %T", typeOrData),
```

Comment:
This message is now misleading as it applies to `OneOfSchema*` not just `AnySchema`

But I think I also need some explanation of exactly how this change (essentially "accepting one-of as 'any'") provides the result advertised in the PR description? 😦

Reply:
I improved the documentation and the tests.

Reply:
The comment is good, but the "'any' type" in the error message still seems a bit misleading.